### PR TITLE
feat(components): add insertion tab for mutations component

### LINKED ISSUE #106: Mutations: Add a tab for insertions table

### LINKED ISSUE #107: Mutations Table: Allow users to sort by any of the columns
Mutations table with nucleotides

### PR DESCRIPTION
**Pull request by JonasKellerer**
resolves #106, resolves #107

### Summary

Adds a new tab on the mutations component to show insertions. The download button now downloads either the insertions, or the substitutions/deletions.

Also in this PR: 
- sorting for all table columns now in one place
- move query for insertion to its own query

### Screenshot

grafik

### PR Checklist

- [x] The implemented feature is covered by an appropriate test.

### CODE REVIEW

File: components/src/preact/shared/sort/sortInsertions.spec.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    test('should sort for symbols second', () => {
+        const a = 'ins_1:A';
+        const b = 'ins_1:B';
+
+        expect(sortInsertions(a, b)).toBeLessThan(0);
+        expect(sortInsertions(b, a)).toBeGreaterThan(0);
+    });
+
+    test('should sort for prefix third', () => {
```
Comment (fengelniederhammer) [inline components/src/preact/shared/sort/sortInsertions.spec.ts:22]: The title doesn't really fit, as it doesn't sort.

File: components/src/query/queryMutations.ts
Comment (fengelniederhammer) [inline components/src/query/queryMutations.ts:1]: This file is empty.

**Review by fengelniederhammer — APPROVED** [review]
Nice :+1:

File: components/src/operator/FetchSubstitutionsOrDeletionsOperator.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        for (const { mutation, count, proportion } of mutations) {
+            const parsed = MutationCache.getInstance().getMutation(mutation);
+            if (parsed instanceof Substitution) {
+                content.push({ type: 'substitution', mutation: parsed, count, proportion });
+            } else if (parsed instanceof Deletion) {
+                content.push({ type: 'deletion', mutation: parsed, count, proportion });
+            } else {
+                throw new Error('Unexpected mutation type');
+            }
+        }
```
[the remark this answers is not in the document]
Comment (JonasKellerer) [reply components/src/operator/FetchSubstitutionsOrDeletionsOperator.ts:28]: After we migrated to LAPIS2 we can remove the mutation cache. #123. Therefore, I leave it in until then.

**Pull request merged**